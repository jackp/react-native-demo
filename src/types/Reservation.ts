export interface Reservation {
  id: string;
  name: string;
  hotelName: string;
  arrivalDate: string;
  departureDate: string;
}
