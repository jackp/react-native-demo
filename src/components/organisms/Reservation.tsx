/**
 * <Reservation />
 * - Display reservation information
 */

import React from "react";
import { Card, CardItem, Text, Title, Body } from "native-base";

import { Reservation as ReservationType } from "../../types/Reservation";

interface IProps {
  reservation: ReservationType;
}

const Reservation: React.FunctionComponent<IProps> = ({ reservation }) => (
  <Card transparent={true} style={{ flex: 1 }}>
    <CardItem header={true}>
      <Title testID="name">{reservation.name}</Title>
    </CardItem>
    <CardItem>
      <Body>
        <Text testID="hotelName">Hotel: {reservation.hotelName}</Text>
        <Text testID="arrivalDate">
          Arrival Date: {reservation.arrivalDate}
        </Text>
        <Text testID="departureDate">
          Departure Date: {reservation.departureDate}
        </Text>
      </Body>
    </CardItem>
  </Card>
);

export default Reservation;
