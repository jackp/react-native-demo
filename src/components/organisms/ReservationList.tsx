/**
 * <ReservationList />
 * - Display list of reservations
 */

import React from "react";
import { List, ListItem } from "native-base";

import Reservation from "./Reservation";

import { Reservation as ReservationType } from "../../types/Reservation";

interface IProps {
  reservations: ReservationType[];
}

const ReservationList: React.FunctionComponent<IProps> = ({ reservations }) => (
  <List>
    {reservations.map(reservation => (
      <ListItem key={reservation.id}>
        <Reservation reservation={reservation} />
      </ListItem>
    ))}
  </List>
);

export default ReservationList;
