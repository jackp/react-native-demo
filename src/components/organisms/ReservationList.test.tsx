import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { Reservation as ReservationType } from "../../types/Reservation";
import ReservationList from "./ReservationList";

const reservations: ReservationType[] = [
  {
    id: "1",
    name: "Jack Parker",
    hotelName: "Hilton PVD",
    arrivalDate: "3/7/2019",
    departureDate: "3/17/2019"
  },
  {
    id: "2",
    name: "Bob Barker",
    hotelName: "Hilton Las Vegas",
    arrivalDate: "3/17/2019",
    departureDate: "6/17/2019"
  },
  {
    id: "3",
    name: "George Costanza",
    hotelName: "Hilton NYC",
    arrivalDate: "4/7/2019",
    departureDate: "4/17/2019"
  }
];

describe("<ReservationList />", () => {
  it("renders a null state if there are no reservations", () => {
    const wrapper = shallow(<ReservationList reservations={[]} />);

    expect(wrapper.render());
  });

  it("renders all reservations", () => {
    const wrapper = shallow(<ReservationList reservations={reservations} />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper).toContainMatchingElements(3, "Reservation");
  });
});
