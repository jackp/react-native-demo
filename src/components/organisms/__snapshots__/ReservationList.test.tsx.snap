// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReservationList /> renders all reservations 1`] = `
<Styled(List)>
  <Styled(ListItem)
    key="1"
  >
    <Reservation
      reservation={
        Object {
          "arrivalDate": "3/7/2019",
          "departureDate": "3/17/2019",
          "hotelName": "Hilton PVD",
          "id": "1",
          "name": "Jack Parker",
        }
      }
    />
  </Styled(ListItem)>
  <Styled(ListItem)
    key="2"
  >
    <Reservation
      reservation={
        Object {
          "arrivalDate": "3/17/2019",
          "departureDate": "6/17/2019",
          "hotelName": "Hilton Las Vegas",
          "id": "2",
          "name": "Bob Barker",
        }
      }
    />
  </Styled(ListItem)>
  <Styled(ListItem)
    key="3"
  >
    <Reservation
      reservation={
        Object {
          "arrivalDate": "4/7/2019",
          "departureDate": "4/17/2019",
          "hotelName": "Hilton NYC",
          "id": "3",
          "name": "George Costanza",
        }
      }
    />
  </Styled(ListItem)>
</Styled(List)>
`;
